Task 3.7 Created a User Profile View and implemented State-Based Routing & Navigation

saragrosser opened the pull request:
I have installed react-router and react-router-dom packages by running npm install react-router react-router-dom --save.
Refactored the code in render() method in main-view.jsx to use a BrowserRouter, and add routes to the views.
Replaced buttons that use onMovieClick and onBackClick with React Router links. 

Also created a profile-view folder with UserInfo, UpdateUser, FavouriteMovies and the profile-view.jsx files

saragrosser committed:
Implemented State-Based Routing & Navigation

saragrosser committed:
Created a User Profile View

truetechcode commented on src/Components/navigation-bar/navigation-bar.jsx:
Add a link to your profile here.

truetechcode commented on src/Components/main-view/main-view.jsx:
```diff
@@ -1,103 +1,152 @@
 import Button from "react-bootstrap/Button";
+import { BrowserRouter, Routes, Route, Navigate } from "react-router-dom";
+import { ProfileView } from "../profile-view/profile-view.jsx";
+import "./main-view.scss";
```
You added this line, without adding the file. Remove it if you don't yet have the file

truetechcode commented on src/Components/movie-view/movie-view.jsx:
```diff
@@ -1,11 +1,17 @@
-import Button from "react-bootstrap/Button";
+import { useParams } from "react-router-dom";
+import { Link } from "react-router-dom";
+import "./movie-view.scss";
```
You added this line, without adding the file. Remove it if you don't yet have the file

truetechcode commented on src/Components/profile-view/profile-view.jsx:
The profile view does not yet work

truetechcode commented on src/Components/main-view/main-view.jsx:
You need to list some movies here for test purposes.

truetechcode reviewed (changes requested):
I can now create a user account, but I am unable to login, and check that the spelling of `UserName` is as expected on the backend. 

![](https://coach-courses-us.s3.amazonaws.com/comments/users/40649/3ef8a8df2a3db657446c304324745081/Screenshot-2024-04-18-020959.png)

saragrosser committed:
Troubleshooting

Fixed the errors for login in and creating a new account (sign up)

truetechcode commented on public/index.html:
Are you able to start your app with this change, I was unable to run your app because of this change, you should revert this to what it was in the last PR, and also put it back in the `src` directory.

Let me know if you have a reason why this should remain.

saragrosser replied:
I had to move it to public, because I was unable to run my app without this change

truetechcode replied:
Ok, that's fine, it looks like an issue with parcel, I will update the version of parcel on my PC.

truetechcode commented on src/Components/navigation-bar/navigation-bar.jsx:
Add a link to your profile page in the navigation for authenticated users.

truetechcode reviewed (changes requested):
Hi Sara,

Your login now works, great work on that. Looking at the items in part two of this exercise, I've listed some of the items you will need to work on below. 

> Add a user Profile view **(linked to the main navigation menu for authenticated users)**.
Add a “Favorite” button to your MovieCard and/or MovieView components, so that logged in users can select a movie to store in their list of favorites;
Allow a user to remove a movie from their list of favorites.
Test that all the links in the navigation bar are working correctly. For example, can a logged in user navigate to their profile?

For the favorite movies feature to work as described above you need some movies in your database.
- First, you should try using Postman to pull all movies from your API, as you did in the previous achievement (try querying your database from the terminal also).
- If NO movies are returned in Postman, you likely don't have any movies saved in your database, you will have to follow the instructions in the previous achievement to populate your database with movies and go over the process again.

These are the requirements for this task to be completed.

Best,
Terver

saragrosser committed:
Made corrections to Task 3.7

Updated profile-view, update-suer movie-card, and movie-view files

truetechcode reviewed (approved):
While you will be proceeding with the rest of the achievement, take some time and go over this achievement.

I also noticed that why I wasn't able to run your app was because you are not using Parcel as the app Bundler.

saragrosser committed:
Moved from CRA to Parcel

Made corrections to run the app on Parcel

saragrosser committed:
updated movie-view propTypes

saragrosser committed:
Corrected undefined movie ID

Corrected the error "undefined" id in movies, changed movie-view and main-view to matched

saragrosser committed:
Styling and corrections
